Automattic/jetpack
# Pay with PayPal: Use post GUID as `postLinkUrl` attribute

Fixes https://github.com/Automattic/jetpack/issues/20724

#### Changes proposed in this Pull Request:

Updates the `postLinkUrl` attribute of the Pay with PayPal block so it uses the immutable post GUID (which also works as the post URL) to prevent the editor from becoming dirty after publishing a post.

#### Other information:


#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- [Spin up a JN site running the branch of this PR](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=fix/pay-with-paypal-post-url).
- Set up Jetpack with a Security plan.
- Go to Posts > Add new.
- Insert a Pay with PayPal block.
- Publish the post.
- Make sure it doesn't become dirty (there shouldn't be any unsaved change).
- Switch to the code editor.
- Note the href attribute.
- Make sure it matches the permalink plain format (`<DOMAIN>?p=<POST_ID>`).
